File: Table_1.py
# Table 1. Baseline characteristics according to baseline pain severity.
import numpy as np
import pandas as pd
from scipy.stats import mannwhitneyu
from _common import MILD, MODSEV, bh, categorical_p_from_table, fmt_p, read_sheet, save_csv


df = read_sheet("Baseline_191")
group_col = "Baseline_Pain_Group_2L"

continuous = [
    ("Age, years", "Age", 1),
    ("Baseline VAS", "Baseline_VAS", 1),
    ("CMO, mm", "CMO", 1),
    ("MMO, mm", "MMO", 1),
    ("Pain-location burden score", "Pain_Location_Burden_0_6", 1),
]
categorical = [
    ("Female sex, n (%)", "Sex", "F"),
    ("Pain present, n (%)", "Pain_Present", 1),
    ("Rt. Masseter pain, n (%)", "Rt_Masseter", 1),
    ("Rt. Temporalis pain, n (%)", "Rt_Temporalis", 1),
    ("Rt. TMJ pain, n (%)", "Rt_TMJ", 1),
    ("Lt. Masseter pain, n (%)", "Lt_Masseter", 1),
    ("Lt. Temporalis pain, n (%)", "Lt_Temporalis", 1),
    ("Lt. TMJ pain, n (%)", "Lt_TMJ", 1),
]


def cont_cell(s):
    x = pd.Series(s).dropna().astype(float)
    return f"{x.mean():.1f} ± {x.std(ddof=1):.1f} ({len(x)})"


def cat_cell(s, positive):
    s = pd.Series(s).dropna()
    n = int((s == positive).sum())
    return f"{n} ({100*n/len(s):.1f})"

rows = []
raw_p = []
meta = []

for label, var, _ in continuous:
    a = df.loc[df[group_col] == MILD, var].dropna()
    b = df.loc[df[group_col] == MODSEV, var].dropna()
    p = mannwhitneyu(a, b, alternative="two-sided").pvalue
    raw_p.append(p)
    meta.append(("continuous", label, var, None))

for label, var, positive in categorical:
    x = (df[var] == positive) if var == "Sex" else df[var]
    tab = pd.crosstab(x, df[group_col])
    p = categorical_p_from_table(tab)
    raw_p.append(p)
    meta.append(("categorical", label, var, positive))

adj = bh(raw_p)

for i, (kind, label, var, positive) in enumerate(meta):
    if kind == "continuous":
        overall = cont_cell(df[var])
        mild = cont_cell(df.loc[df[group_col] == MILD, var])
        modsev = cont_cell(df.loc[df[group_col] == MODSEV, var])
    else:
        overall = cat_cell(df[var], positive)
        mild = cat_cell(df.loc[df[group_col] == MILD, var], positive)
        modsev = cat_cell(df.loc[df[group_col] == MODSEV, var], positive)
    rows.append({
        "Variable": label,
        "Overall": overall,
        "Mild pain (VAS 0–3)": mild,
        "Moderate-to-severe pain (VAS ≥4)": modsev,
        "P value": fmt_p(raw_p[i]),
        "FDR-adjusted P value": fmt_p(adj[i]),
    })

patients = pd.DataFrame([{
    "Variable": "Patients, n",
    "Overall": str(len(df)),
    "Mild pain (VAS 0–3)": str((df[group_col] == MILD).sum()),
    "Moderate-to-severe pain (VAS ≥4)": str((df[group_col] == MODSEV).sum()),
    "P value": "",
    "FDR-adjusted P value": "",
}])

table = pd.concat([patients, pd.DataFrame(rows)], ignore_index=True)
save_csv(table, "Table_1.csv")
print(table.to_string(index=False))

File: Table_1.R
suppressPackageStartupMessages({library(readxl);library(dplyr)})
source(file.path(dirname(sub("^--file=", "", commandArgs(FALSE)[grepl("^--file=", commandArgs(FALSE))][1])), "_common.R"))
dat <- read_excel(resolve_data_path(), sheet = "Baseline_191")
cont <- c("Age","Baseline_VAS","CMO","MMO","Pain_Location_Burden_0_6")
cont_labels <- c("Age, years","Baseline VAS","CMO, mm","MMO, mm","Pain-location burden score")
cat_vars <- c("Sex","Pain_Present","Rt_Masseter","Rt_Temporalis","Rt_TMJ","Lt_Masseter","Lt_Temporalis","Lt_TMJ")
cat_labels <- c("Female sex, n (%)","Pain present, n (%)","Rt. Masseter pain, n (%)","Rt. Temporalis pain, n (%)","Rt. TMJ pain, n (%)","Lt. Masseter pain, n (%)","Lt. Temporalis pain, n (%)","Lt. TMJ pain, n (%)")

ps <- c()
for (v in cont) ps <- c(ps, wilcox.test(dat[[v]] ~ dat$Baseline_Pain_Group_2L, exact=FALSE)$p.value)
for (v in cat_vars) {
  x <- if (v == "Sex") dat[[v]] == "F" else dat[[v]]
  ps <- c(ps, categorical_p(table(x, dat$Baseline_Pain_Group_2L)))
}
adj <- p.adjust(ps, method="BH")

rows <- list(); k <- 1
cell_cont <- function(x) {x<-x[!is.na(x)]; sprintf("%.1f ± %.1f (%d)",mean(x),sd(x),length(x))}
cell_cat <- function(x,pos) {x<-x[!is.na(x)]; n<-sum(x==pos); sprintf("%d (%.1f)",n,100*n/length(x))}
for(i in seq_along(cont)){
 v<-cont[i]; rows[[length(rows)+1]]<-data.frame(Variable=cont_labels[i],Overall=cell_cont(dat[[v]]),Mild=cell_cont(dat[[v]][dat$Baseline_Pain_Group_2L==MILD]),Moderate_to_severe=cell_cont(dat[[v]][dat$Baseline_Pain_Group_2L==MODSEV]),P_value=fmt_p(ps[k]),FDR_adjusted_P_value=fmt_p(adj[k]),check.names=FALSE);k<-k+1
}
for(i in seq_along(cat_vars)){
 v<-cat_vars[i]; pos<-if(v=="Sex") "F" else 1; rows[[length(rows)+1]]<-data.frame(Variable=cat_labels[i],Overall=cell_cat(dat[[v]],pos),Mild=cell_cat(dat[[v]][dat$Baseline_Pain_Group_2L==MILD],pos),Moderate_to_severe=cell_cat(dat[[v]][dat$Baseline_Pain_Group_2L==MODSEV],pos),P_value=fmt_p(ps[k]),FDR_adjusted_P_value=fmt_p(adj[k]),check.names=FALSE);k<-k+1
}
tab <- bind_rows(data.frame(Variable="Patients, n",Overall=nrow(dat),Mild=sum(dat$Baseline_Pain_Group_2L==MILD),Moderate_to_severe=sum(dat$Baseline_Pain_Group_2L==MODSEV),P_value="",FDR_adjusted_P_value="",check.names=FALSE), bind_rows(rows))
write.csv(tab,file.path(output_dir(),"Table_1.csv"),row.names=FALSE,fileEncoding="UTF-8")
print(tab)
